### PR TITLE
[REF] split beesdoo_pos

### LINKED ISSUE #344: Split `beesdoo_pos` into `pos_eater`

### LINKED ISSUE #345: Split `beesdoo_pos` into `pos_unavailable_for_sale`
According to https://github.com/coopiteasy/vertical-cooperative/issues/204 'better name should be found'.

### PR DESCRIPTION
**Pull request by victor-champonnois**
## Description
fix https://github.com/beescoop/Obeesdoo/issues/344
fix https://github.com/beescoop/Obeesdoo/issues/345

related PR :
- https://github.com/coopiteasy/cie-custom/pull/88
- https://github.com/coopiteasy/cie-pos/pull/23
- https://github.com/OCA/pos/pull/879
- https://github.com/OCA/pos/pull/880
------------------------

Question : what to do with the JS and CSS that is still in beesdoo_pos. It doesn''t seem to be related to the splitted functionalities. In fact the remaining functionalities are
```
- hides phone and address field in pos customer list
- do not allow customer creation from the POS
```
Should I put them in a bees_custom module ?
@robinkeunen @remytms 

Also the obeesdoo refactoring plans says that a better name should be found for `pos_unavailable_for_sale`. Don't have an idea yet, anyone has one ?

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

File: pos_eater/models/res_partner.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 from odoo import api, models
 
 
-class BeescoopPosPartner(models.Model):
+class ResPartner(models.Model):
     _inherit = "res.partner"
 
     @api.multi
```
Comment (robinkeunen) [inline pos_eater/models/res_partner.py:7]: The javascript files were not moved to pos_eater.

**Review by robinkeunen — CHANGES_REQUESTED** [review]
> Question : what to do with the JS and CSS that is still in beesdoo_pos. It doesn''t seem to be related to the splitted functionalities. In fact the remaining functionalities are
> 
> ```
> - hides phone and address field in pos customer list
> - do not allow customer creation from the POS
> ```

I think both remaining features would deserve an oca/pos module actually.

> Also the obeesdoo refactoring plans says that a better name should be found for `pos_unavailable_for_sale`. Don't have an idea yet, anyone has one ?

Tricky... pos_available_in_pos_unlink? 

Or maybe question the need for this functionality altogether ? I can send an email to community@komunigi.org

- if no one wants it, remove it,
- if only the bees wants to keep it, let's move it to beesdoo_custom,
- if more want this behavior, ask a name advice to Sylvain

**Comment by victor-champonnois** [comment]
@robinkeunen I extracted the two functionalities in separate modules : `pos_forbid_new_customer` et `pos_hide_customer_info`. 

There are still some remaining stuff in beesdoo_pos:
- some css
- A JS function that makes an invoice when changing customer in the payment screen (?) There is a [pos_auto_invoice](https://github.com/coopiteasy/addons/tree/12.0/pos_auto_invoice) modules in addons but it doesn't seem related (it's only installed in rotordc)
- A default barcode_rule_client (?)

An idea what to do with those ?

---------------------------------

> Tricky... pos_available_in_pos_unlink?
> 
> Or maybe question the need for this functionality altogether ? I can send an email to [community@komunigi.org](mailto:community@komunigi.org)
> 
>     * if no one wants it, remove it,
> 
>     * if only the bees wants to keep it, let's move it to beesdoo_custom,
> 
>     * if more want this behavior, ask a name advice to Sylvain

OK let's do that, although, unless nobody wants it I would still keep it in its own module, even with a bad name, since we already did the work of unbranding it. Don't you think ?

**Comment by robinkeunen** [comment]
> @robinkeunen I extracted the two functionalities in separate modules : `pos_forbid_new_customer` et `pos_hide_customer_info`.

👌 

> There are still some remaining stuff in beesdoo_pos:
> 
> * some css
> * A JS function that makes an invoice when changing customer in the payment screen (?) There is a [pos_auto_invoice](https://github.com/coopiteasy/addons/tree/12.0/pos_auto_invoice) modules in addons but it doesn't seem related (it's only installed in rotordc)

I'd say `pos_auto_invoice_company`. Rotor's module applies `to_invoice` to all orders and this module only on company partners.

> * A default barcode_rule_client (?)

Boh... bees_custom ? @remytms where can we put this ?

```xml
<odoo>
    <data>
        <record id="point_of_sale.barcode_rule_client" model="barcode.rule">
            <field name="encoding">ean13</field>
        </record>
    </data>
</odoo>
```

> OK let's do that, although, unless nobody wants it I would still keep it in its own module, even with a bad name, since we already did the work of unbranding it. Don't you think ?

Of course 👌

File: pos_hide_customer_info/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+This module :
+- hides phone and address field in pos customer list
+- do not allow customer creation from the POS
```
Comment (robinkeunen) [inline pos_hide_customer_info/readme/DESCRIPTION.rst]: this line got lost.

File: beesdoo_pos/__manifest__.py [hunk 3]
```diff
@@ -9,13 +9,12 @@
...
     "name": "BEES coop Point of sale",
-    "summary": """This module adds the eaters of the customer to the POS
-    ActionpadWidget and PaymentScreenWidget.""",
+    "summary": """Emptied.""",
     "author": "BEES coop - Cellule IT, Coop IT Easy SC",
     "website": "https://github.com/beescoop/Obeesdoo",
     "category": "Point Of Sale",
     "version": "12.0.2.0.0",
-    "depends": ["eater", "beesdoo_product"],
+    "depends": ["point_of_sale", "pos_eater", "pos_unavailable_for_sale"],
```
Comment (robinkeunen) [inline beesdoo_pos/__manifest__.py]: point_of_sale is not needed.

File: pos_eater/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+#   - Elouan Lebars <elouan@coopiteasy.be>
+#   - Rémy Taymans <remy@coopiteasy.be>
+#   - Vincent Van Rossem <vincent@coopiteasy.be>
+#   - Elise Dupont
+#   - Thibault François
+#   - Grégoire Leeuwerck
+#   - Houssine Bakkali <houssine@coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "BEES coop Eater",
```
Comment (robinkeunen) [inline pos_eater/__manifest__.py]: Rename to `POS Eater`

File: pos_forbid_new_customer/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1,3 @@
+This module :
+- hides phone and address field in pos customer list
```
Comment (robinkeunen) [inline pos_forbid_new_customer/readme/DESCRIPTION.rst]: This line belongs to the other module.

**Review by robinkeunen — CHANGES_REQUESTED** [review]
@victor-champonnois en conclusion des échanges de mail : 

- `barcode_rule_client` => dans bees_custom
- `pos_available_in_pos_unlink` leave in this module, move to cie-pos
- `pos_forbid_new_customer` => to oca/pos 
  - with name suggestion : `pos_partner_no_create` (or ask sylvain for a better name)
- `pos_hide_customer_info` => to oca/pos
  - with name pos_hide_partner_info

File: pos_hide_customer_info/readme/DESCRIPTION.rst (hunk 2 above)
Comment (victor-champonnois) [reply pos_hide_customer_info/readme/DESCRIPTION.rst]: fixed

File: pos_forbid_new_customer/readme/DESCRIPTION.rst (hunk 5 above)
Comment (victor-champonnois) [reply pos_forbid_new_customer/readme/DESCRIPTION.rst]: fixed

File: pos_eater/__manifest__.py (hunk 4 above)
Comment (victor-champonnois) [reply pos_eater/__manifest__.py]: fixed

File: beesdoo_pos/__manifest__.py (hunk 3 above)
Comment (victor-champonnois) [reply beesdoo_pos/__manifest__.py]: fixed

**Comment by victor-champonnois** [comment]
@robinkeunen Ready to review again :)

**Comment by victor-champonnois** [comment]
tests fail becaus it depends on other unmerged PR

File: beesdoo_pos/__manifest__.py [hunk 6]
```diff
@@ -9,15 +9,17 @@
...
     "website": "https://github.com/beescoop/Obeesdoo",
     "category": "Point Of Sale",
     "version": "12.0.2.0.0",
-    "depends": ["eater", "beesdoo_product"],
-    "data": ["views/beesdoo_pos.xml", "data/default_barcode_pattern.xml"],
-    "qweb": ["static/src/xml/templates.xml"],
+    "depends": [
+        "pos_eater",
+        "pos_available_in_pos_unlink",
+        "pos_forbid_new_customer",
```
Comment (robinkeunen) [inline beesdoo_pos/__manifest__.py]: @victor-champonnois is this another module than `pos_partner_no_create`?

File: beesdoo_pos/__manifest__.py [hunk 7]
```diff
@@ -9,15 +9,17 @@
...
     "author": "BEES coop - Cellule IT, Coop IT Easy SC",
     "website": "https://github.com/beescoop/Obeesdoo",
     "category": "Point Of Sale",
     "version": "12.0.2.0.0",
-    "depends": ["eater", "beesdoo_product"],
-    "data": ["views/beesdoo_pos.xml", "data/default_barcode_pattern.xml"],
-    "qweb": ["static/src/xml/templates.xml"],
+    "depends": [
+        "pos_eater",
+        "pos_available_in_pos_unlink",
```
Comment (robinkeunen) [inline beesdoo_pos/__manifest__.py:19]: plutot pos_unavailable_for_sale

File: beesdoo_pos/__manifest__.py (hunk 6 above)
Comment (victor-champonnois) [reply beesdoo_pos/__manifest__.py]: No it's the same

**Comment by victor-champonnois** [comment]
Status : this depends on https://github.com/OCA/pos/pull/880 that needs to be done before merging this PR

**Review by robinkeunen — APPROVED** [review]

**Pull request merged**